### PR TITLE
feat: cache build in tiflash pipeline

### PR DESCRIPTION
**Pull request by purelind**
For PRs with significant changes, tiflash build binary generally takes 25 minutes. For some test instabilities or environment issues causing pipeline execution failures, repeatedly building the same commit SHA binary of a PR is unnecessary; therefore, adding a build cache is used for acceleration.

### CODE REVIEW

File: pipelines/pingcap/tiflash/latest/pull_unit_test.groovy [hunk 1]
```diff
@@ -50,9 +51,32 @@ pipeline {
                         currentBuild.description = "PR #${REFS.pulls[0].number}: ${REFS.pulls[0].title} ${REFS.pulls[0].link}"
                     }
                 }
+                script {
+                    // test build cache, if cache is exist, then skip the following build steps
+                    try {
+                        dir("test-build-cache") { 
+                            cache(path: "./", includes: '**/*', key: prow.getCacheKey('git', REFS, 'ut-build')){
```
Comment (wuhuizuo) [inline pipelines/pingcap/tiflash/latest/pull_unit_test.groovy:58]: prow.getCacheKey('tiflash-ut-build', ...

**Review by wuhuizuo — APPROVED** [review]

**Review by Lloyd-Pottiger — APPROVED** [review]
👍  lgtm

**Pull request merged**